std/range/package.d

```diff
@@ -9213,3 +9221,292 @@ if (is(typeof(fun) == void) || isSomeFunction!fun)
 
     auto r = [1, 2, 3, 4].tee!func1.tee!func2;
 }
+
+/**
+Extend the length of the input range $(D r) by padding out the start of the
+range with the element $(D e). The element $(D e) must be of a common type with
+the element type of the range $(D r) as defined by $(REF CommonType, std, traits).
+If $(D n) is less than the length of of $(D r), then the contents of $(D r) are
+returned.
+
+If $(D r) is a string with Unicode characters in it, remember D's rules about the
+length of strings: length for strings is not the number of characters, or
+graphemes, but instead the number of encoding units. If you want to treat each
+grapheme as only one encoding unit long, then call
+$(REF byGrapheme, std, uni) before calling this function.
+
+If $(D r) is an input range, this function is $(BIGOH 1). But if it is a forward
+range, it's $(BIGOH r.length).
```

Comment:
I think the correct formulation here is "Complexity is $(BIGOH 1) if $(D r) has length, $(BIGOH n) otherwise. Input ranges without length are not usable with padLeft."